src/utils/utils.js

```diff
@@ -501,7 +534,158 @@ var utils = {
             arr.push(i);
         }
         return arr;
+    },
+
+    flatten: (array) => {
+        if (!Array.isArray(array)) {
+            return array;
+        }
+        return [].concat(...array.map(x => utils.flatten(x)));
+    },
+
+    unique: (array, func) => {
+        let mappedArray = array;
+        let filter = ((elem, pos, arr) => arr.indexOf(elem) === pos);
+        if (typeof func === 'function') {
+            mappedArray = array.map(func);
+            filter = (elem, pos) => {
+                let mappedElem = mappedArray[pos];
+                return mappedArray.findIndex(x => x === mappedElem) === pos;
+            };
+        }
+        return array.filter(filter);
+    },
+
+    groupBy: (array, func) => {
+        return array.reduce((obj, v) => {
+            var group = func(v);
+            obj[group] = obj[group] || [];
+            obj[group].push(v);
+            return obj;
+        }, {});
+    },
+
+    union: (arr1, arr2) => utils.unique(arr1.concat(arr2)),
+
+    intersection: (arr1, arr2) => arr1.filter(x => arr2.indexOf(x) !== -1),
+
+    defaults: (obj, ...defaultObjs) => {
+        var length = defaultObjs.length;
+        if (length === 0 || !obj) {
+            return obj;
+        }
+        for (var index = 0; index < length; index++) {
+            var source = defaultObjs[index],
+                keys = utils.isObject(source) ? Object.keys(source) : [],
+                l = keys.length;
+            for (var i = 0; i < l; i++) {
+                var key = keys[i];
+                if (obj[key] === undefined) {
+                    obj[key] = source[key];
+                }
+            }
+        }
+        return obj;
+    },
+
+    omit: (obj, prop) => {
```

Comment:
You mean that is can also take a function or several properties?

I've simplified it because it's just use for simple properties in the tauCharts.
Do you prefer it to be compliant with underscore even if most of the functionality is unused?